Show location in the settings xblock.

oksana-slu opened the pull request:
Background: 
Review of xblock location is enabled. Field "display_block_location" is added to course settings; this is a field visibility of xblock location depends on. This functionality is an important part of [Conditional Mode](https://github.com/edx/edx-platform/pull/11710) editing 

Studio Updates: 
The next file is modified: common/lib/xmodule/xmodule/course_module.py, common/lib/xmodule/xmodule/x_module.py.
The new tests are added common/test/acceptance/tests/studio/test_studio_container.py, common/lib/xmodule/xmodule/tests/test_course_module.py. 
For new functionality, the next tests are corrected: common/lib/xmodule/xmodule/tests/test_editing_module.py, common/lib/xmodule/xmodule/tests/test_split_test_module.py, common/lib/xmodule/xmodule/tests/test_xml_module.py, common/test/acceptance/pages/studio/settings_advanced.py, lms/djangoapps/courseware/tests/test_video_mongo.py.

LMS Updates: None

![img 1](https://dl.dropboxusercontent.com/u/28163465/location_2.png)

![img 2](https://dl.dropboxusercontent.com/u/28163465/location_1.png)

edx-webhook commented:
Thanks for the pull request, @oksana-slu! I've created [OSPR-1217](https://openedx.atlassian.net/browse/OSPR-1217) to keep track of it in JIRA. JIRA is a place for product owners to prioritize feature reviews by the engineering development teams.

Feel free to add as much of the following information to the ticket:
- supporting documentation
- edx-code email threads
- timeline information ("this must be merged by XX date", and why that is)
- partner information ("this is a course on edx.org")
- any other information that can help Product understand the context for the PR

All technical communication about the code itself will still be done via the GitHub pull request interface. As a reminder, [our process documentation is here](http://edx-developer-guide.readthedocs.org/en/latest/process/overview.html).



If you like, you can add yourself to the [AUTHORS](https://github.com/edx/edx-platform/blob/master/AUTHORS)      file for this repo, though that isn't required.     Please see the [CONTRIBUTING](https://github.com/edx/edx-platform/blob/master/CONTRIBUTING.rst)     file for more information.

nedbat commented:
@oksana-slu I don't understand what this means.  Maybe @scottrish has some context.

oksana-slu commented:
@nedbat @scottrish 
We have developed conditional mode which is enabled within Studio functionality. For using new conditional mode user needs to add location in source list (conditional setting, see picture ).  
![img 4](https://dl.dropboxusercontent.com/u/28163465/conditional_4.png)
At the moment it is quite difficult to get this data in Open edX. So we add to advanced settings new field "Display block location". If user sets "true", every xblock will contain location (read only). This data can be added in settings of conditional mode.

marcotuts commented:
This setting doesn't seem like something authors should need to set. What is the benefit of the conditional logic / flag to exposing the course ID / location? Could this be always included behind the scenes to an xblock? Is there any downside to doing this? 

I ask because the expectation is that settings in the Settings tab for an XBlock should (whenever possible) be meaningful to course authors and not be configuration logic for xblock details.

jbarciauskas commented:
@oksana-slu any thoughts based on @marcotuts 's feedback?

marcotuts commented:
Ah! Thanks for the update on this joel. @oksana-slu sent me a document via email that has answers, I will reply to that and add a link to this thread for next steps. I think this might be back in 'product review'

nedbat commented:
@oksana-slu are you still interested in pursuing this?

oksana-slu commented:
@nedbat 
Hello!
Sorry for delaying.(I miss your comment). We are still interested in this PR.
It could be rendered as a button, but we need time to change the business logic of location feature. 
If we return to this in month, will it be ok? 
Does this PR influence on [https://github.com/edx/edx-platform/pull/11710](url)?
thank you

oksana-slu commented:
@marcotuts @nedbat 
Could I ask a question.
I adapted source code according the requirements.
But taking into account your comments, my previous commit is not of current interest. Should I make new PR or just change the source code in current PR.
Thanks

jbarciauskas commented:
@oksana-slu it's up to you - if the historical context is useful, then we should continue on this PR. If it's more helpful to restart the discussion, then close and create a new one

nedbat commented:
@oksana-slu Feel free to ping people here in a comment when you are ready for more review.

oksana-slu commented:
@nedbat 
I have updated PR with your comments, review it please.

cahrens commented:
Note that tests are not running on this PR, @nedbat

benpatterson commented:
Can one of the admins verify this patch?

cahrens commented on cms/templates/js/metadata-editor.underscore:
```diff
@@ -3,4 +3,10 @@
     <% }) %>
+    <li class="field comp-setting-entry metadata_entry">
+        <div class="wrapper-comp-setting">
+            <label class="label setting-label"><%= gettext("Location id") %></label>
```
Should be using <%- instead of <%= (both this line and the next) to ensure the text is escaped.

jbarciauskas commented:
Jenkins ok to test

jbarciauskas commented:
Jenkins add to whitelist

oksana-slu commented:
@cahrens 
I fixed everything what you asked.
Thanks

jbarciauskas commented:
jenkins run all

nedbat commented:
Jenkins ok to test

jbarciauskas commented:
jenkins run all

jbarciauskas commented:
@oksana-slu looks like there's a build error - can you look into it?

cahrens commented:
FYI that the safe templating change was just something I noticed while looking over OSPRs-- I'm not actually reviewing this PR. We still need product sign-off on this feature.

shaunagm commented:
jenkins run all

shaunagm commented:
@marcotuts Can you do a review of @oksana-slu's changes?  Once we get your sign off, we can work on getting those tests to pass.

nedbat commented:
@oksana-slu looks like this needs to be rebased for the tests to run.

marcotuts commented:
Hi! Is there a sandbox I can review this on? I checked the one linked from the other PR and it wasn't working for me. (http://178.62.209.246:18010/) Thanks!

oksana-slu commented:
@marcotuts 
Hi, unfortunately this sandbox doesn't cover the update location feature.
We will deploy the other sandbox for your review in several days.
ASAP we do it, I let you know.
Thanks

oksana-slu commented:
@marcotuts 
We updated the sandbox with location_id
LMS:  https://edx-show-location-id.raccoongang.com
CMS: https://studio-show-location-id.raccoongang.com

marcotuts commented:
Hi @oksana-slu - I'm not seeing the advanced setting called "Display block location" on that sandbox. Is that what I should be looking for? Thanks

cahrens commented:
@marcotuts I believe we asked them to ditch the Advanced Setting. I don't know why you would not always want that information available.

You can see the block location by looking at the settings for an individual component on a unit page. I think some help text explaining what this is would be nice, and I'm not sure I like the fact it is always last (out of alphabetical order).

marcotuts commented:
Thanks for the changes here. 👍  from a product standpoint here.

catong commented on cms/templates/js/metadata-editor.underscore:
```diff
@@ -3,4 +3,10 @@
     <% }) %>
+    <li class="field comp-setting-entry metadata_entry">
+        <div class="wrapper-comp-setting">
+            <label class="label setting-label"><%- gettext("Location id") %></label>
```
Please make "id" uppercase: "Location ID"

catong commented:
@cahrens Also wondering if you think line wrap could/should be improved. Currently when I reduce the size of my browser window the location ID wraps like this (see screenshot).

![locationidlinewrap](https://cloud.githubusercontent.com/assets/9041399/17380096/52973390-5993-11e6-8e34-5fe9e5965790.png)

cahrens commented:
@catong I'm not reviewing this OSPR, but I always like things that are prettier. :)

shaunagm commented:
jenkins run all

shaunagm commented:
@oksana-slu Have you had a chance to rebase your pull request?  Let's try to get those tests passing.

oksana-slu commented:
@shaunagm 
Hello,
We will try to rebase the pull request on the next week.
If it will be ok?
Thanks

shaunagm commented:
That's great, @oksana-slu.  Just ping me when you're done so I can rerun the tests.

shaunagm commented:
@oksana-slu, looks like there was a problem with rebasing?  Let me know if you need help figuring it out.

andy-armstrong commented:
jenkins run all

andy-armstrong commented:
@oksana-slu I'm going to review your PR this week. Can you fix the rebase issue (currently, GitHub sees >250 commits!) and let me know when there is a sandbox available. Thanks!

oksana-slu commented:
@andy-armstrong
I fixed the rebase issue.
We updated the sandbox with location_id 
LMS: https://edx-show-location-id.raccoongang.com
Studio: https://studio-show-location-id.raccoongang.com

One test is failed, I will fix it asap
Thanks

andy-armstrong commented:
jenkins run bokchoy

andy-armstrong commented:
@oksana-slu this looks good to me. 👍 

@marcotuts the changes appear to have addressed your feedback about the rendering of the ID. If you want to give thumbs, then I'll go ahead and merge this.

catong commented:
@andy-armstrong Did "id" in the string get changed to "ID"? 👍 from me if that is done.

catong commented:
Ah, confirmed that it has been updated. Thanks!

marcotuts commented:
Thanks for the updated FYI @andy-armstrong.

👍  from me!

andy-armstrong commented:
@oksana-slu will you have time to fix the Bok Choy test failure this week? It would be great to have this PR merged. Thanks!

oksana-slu commented:
@andy-armstrong 
This test  "test_create_book_message" was ok  locally.
http://prntscr.com/che9zo

Could you please explain what should I do in such situation?
thanks

andy-armstrong commented:
@oksana-slu That's very frustrating, and I don't have much advice to offer. One useful thing is to find the screenshot that is taken when the test fails. In this case it is here:

https://build.testeng.edx.org/job/edx-platform-bok-choy-pr/25747/artifact/test_root/log/shard_2/common.test.acceptance.tests.studio.test_studio_textbooks.TextbooksTest.test_create_first_book_message.png

The test is looking for a message, and the screenshot seems to be showing the correct one, so that doesn't help. Maybe there's a timing problem with the test that isn't introduced by your changes.

Given that your changes seem unrelated, I suggest rebasing again and see if maybe that fixes it. If it doesn't, then I suggest introducing a wait into the test so that it waits for the element to be available.

andy-armstrong commented:
jenkins run bokchoy

catong commented:
@oksana-slu Sorry for this late change. After consulting with @marcotuts we realize we need to change the name of this ID to avoid confusion with the "Location ID" that is visible in the Studio course outline sidebar. 
Can you please change "Location ID" to "Component Location ID"?
Thanks, please let us know if you have any questions. FYI @andy-armstrong

shaunagm commented:
@andy-armstrong @catong @oksana-slu The tests are passing and the ID name change has been made.  Is this ready to be merged?

catong commented:
👍
